x-pack/plugins/fleet/public/applications/fleet/components/form_settings/index.tsx

```diff
@@ -0,0 +1,165 @@
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const validationResults = coercedSchema.safeParse(Number(e.target.value));
+
+    if (!validationResults.success) {
+      setError(validationResults.error.issues[0].message);
+      return;
+    }
+    const newAdvancedSettings = {
+      ...(agentPolicyFormContext?.agentPolicy.advanced_settings ?? {}),
+      [settingsConfig.api_field.name]: parseInt(e.target.value, 10), // TODO support float?
+    };
+
+    agentPolicyFormContext?.updateAgentPolicy({ advanced_settings: newAdvancedSettings });
+    setError('');
+  };
+
+  return (
+    <EuiDescribedFormGroup
+      fullWidth
+      title={<h4>{settingsConfig.title}</h4>}
+      description={
+        <>
+          {settingsConfig.description}.{' '}
+          <EuiLink href={settingsConfig.learnMoreLink} external>
+            Learn more.
+          </EuiLink>
+        </>
+      }
+    >
+      <EuiFormRow fullWidth key={fieldKey} error={error} isInvalid={!!error}>
+        <EuiFieldNumber
+          fullWidth
+          data-test-subj={fieldKey}
+          value={
+            agentPolicyFormContext?.agentPolicy.advanced_settings?.[settingsConfig.api_field.name]
+          }
+          min={coercedSchema.minValue ?? undefined}
+          max={coercedSchema.maxValue ?? undefined}
+          defaultValue={defaultValue}
+          onChange={handleChange}
+          isInvalid={!!error}
+        />
+      </EuiFormRow>
+    </EuiDescribedFormGroup>
+  );
+});
+
+settingComponentRegistry.set(z.string()._def.typeName, (settingsConfig) => {
+  const [error, setError] = useState('');
+  const agentPolicyFormContext = useAgentPolicyFormContext();
+
+  const fieldKey = `configuredSetting-${settingsConfig.name}`;
+  const defaultValue: number =
+    settingsConfig.schema instanceof z.ZodDefault
+      ? settingsConfig.schema._def.defaultValue()
+      : undefined;
+  const coercedSchema = settingsConfig.schema as z.ZodString;
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const validationResults = coercedSchema.safeParse(e.target.value);
+
+    if (!validationResults.success) {
+      setError(validationResults.error.issues[0].message);
+      return;
+    }
+
+    const newAdvancedSettings = {
+      ...(agentPolicyFormContext?.agentPolicy.advanced_settings ?? {}),
+      [settingsConfig.api_field.name]: e.target.value,
+    };
+
+    agentPolicyFormContext?.updateAgentPolicy({ advanced_settings: newAdvancedSettings });
+
+    setError('');
+  };
+
+  return (
+    <EuiDescribedFormGroup
+      fullWidth
+      title={<h4>{settingsConfig.title}</h4>}
+      description={
+        <>
+          {settingsConfig.description}.{' '}
+          <EuiLink href={settingsConfig.learnMoreLink} external>
+            Learn more.
+          </EuiLink>
+        </>
+      }
+    >
+      <EuiFormRow fullWidth key={fieldKey} error={error} isInvalid={!!error}>
+        <EuiFieldText
```

Comment:
there is a lot of duplication between number and text type, I wonder if there could be a common component as a wrapper